geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/Coder.java

```diff
@@ -404,9 +404,14 @@ public static double bytesToDouble(byte[] bytes) {
       return NEGATIVE_INFINITY;
     }
     if (isNaN(bytes)) {
-      return NaN;
+      throw new NumberFormatException();
```

Comment:
Since `Double.NaN` is a thing, it seems inconsistent that we're able to return `*INFINITY`s but throw an exception with `NaN`. Can this exception be thrown, at a higher level where `NaN` is determined to be invalid?

Reply:
Internally, there is no situation in which NaN is ever a valid double value for us. If we ever get passed NaN by the user or would get NaN from a calculation, then we always return an error and never use the actual "NaN" value. As such, it makes sense to just hard-code it into our conversion method rather than adding checks for it all over the place, since there's no situation in which we would have NaN and be okay with it.